cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/store/ProgramScheduleStoreDataset.java

```diff
@@ -162,30 +155,52 @@ public String migrateFromAppMetadataStore(NamespaceId namespaceId, Store appMeta
         }
       }
     }
-    store.put(MIGRATION_COMPLETE_NAMESPACE_ROW_BYTES, MIGRATION_COMPLETE_NAMESPACE_COLUMN_BYTES,
+    store.put(MIGRATION_COMPLETE_STATUS_ROW_BYTES, MIGRATION_COMPLETE_STATUS_COLUMN_BYTES,
               Bytes.toBytes(namespaceId.toString()));
     LOG.info("Schedule migration is completed for namespace '{}'", namespaceId);
-    return namespaceId.toString();
   }
 
-  @Nullable
-  private String getMigrationCompleteNamespace() {
-    Row row = store.get(MIGRATION_COMPLETE_NAMESPACE_ROW_BYTES);
-    if (row.isEmpty()) {
-      return null;
+  /**
+   *
+   */
+  public static class MigrationStatus {
+    private final boolean migrationCompleted;
+    private final NamespaceId lastMigrationCompleteNamespace;
+
+    MigrationStatus(boolean migrationCompleted, @Nullable NamespaceId lastMigrationCompleteNamespace) {
+      this.migrationCompleted = migrationCompleted;
+      this.lastMigrationCompleteNamespace = lastMigrationCompleteNamespace;
+    }
+
+    static MigrationStatus getMigrationCompleteStatus() {
```

Comment:
this can be a constant. And you can also have one for (false, null) to use in line 192. Like
```
static final MigrationStatus MIGRATION_COMPLETE = new MigrationStatus(true, null);
static final MigrationStatus MIGRATION_NOT_STARTED = new MigrationStatus(false, null);
```